### ISSUE TITLE
Trouble with migrations when upgrading from version 3.1.1 to 3.2

### ISSUE THREAD

**Issue by jonathanmorgan** [jdriscoll/django-photologue#134]
I just upgraded django-photologue from version 3.1.1 to version 3.2.  When I run a migrate now, I get the following error message:

django.db.utils.ProgrammingError: (1146, "Table 'mysite.photologue_gallery_sites' doesn't exist")

I have multiple servers that use django-photologue, and I am getting this error everywhere.

Any advice on how I'd fix this?

Thanks,

Jonathan Morgan

**Comment by richardbarran** [comment]
Hi,
Could you please tell me:
- What version of Python you are using.
- What version of Django you are using.
- What database you are using.
- And what version of django-sortedm2m is installed.

Thanks,
Richard

**Comment by jonathanmorgan** [comment]
I have the problem on two separate systems: one is Cent OS 6, up to date, the other is Ubuntu 15.04.

Python version: 2.7.9
Django version: 1.8.2
Database: MySQL 5.1.73 (Cent OS 6) and MariaDB 10.0.17 (ubuntu) - each system has the same behavior.
django-sortedm2m version: 0.10.0

The table the error refers to is not present in either of the databases (photologue_gallery_sites).

If you want more information on the contents of the database (photologue tables present, or contents of the table that stores migration auditing information), let me know.  I could also just make a copy of the database available to you, if that would help.

Also, if it helps, in the past, I’m not sure the version, but before django-photologue implemented django 1.7 migrations, there were messages that I ignored that there were changes to the models in django-photologue that had not yet been turned into migrations.  I generally try to avoid changing migrations for packages I get from pip, so I just ignored those messages, was hoping it would all sort itself out once the package supported new migrations.

Thanks,

Jonathan

-- 

"The man with the new idea is a Crank until the idea succeeds."
- Mark Twain, from 'Following the Equator: A Journey Around the World'

**Comment by jonathanmorgan** [comment]
A clarification - I looked at the migrations install log in one of my databases: the django-photologue migrations are in two clumps - migrations 1 through 4 (applied October of 2014), and then 5 and 6 (applied February of 2015).  I think I've seen messages about changes being missing from migrations after django-photologue added support for django 1.7+ migrations (between October and February), but I am not sure.  Sorry for the uncertainty.

**Comment by jonathanmorgan** [comment]
Another quick note - before doing this manual compare, I tried installing in a clean project and database and running the migrations.  I got the following while it was running migration 0001:

django.db.utils.OperationalError: (1005, 'Can\'t create table `nieonline`.`#sql-6da_28` (errno: 150 "Foreign key constraint is incorrectly formed")')

Tables were created, but the django_migrations table wasn't updated, so in that case, it can't even run the second migration.

I am going to try exactly what you suggested above now, will let you know how it goes.

**Comment by jonathanmorgan** [comment]
Might be related to my tables being MyISAM rather than InnoDB:

https://groups.google.com/forum/#!msg/reviewboard/4gNGVu3Gveg/g34Uj2F5wdoJ

I'm going to try re-importing tables as InnoDB and see if that helps...

**Comment by jonathanmorgan** [comment]
Interesting.  Switching the tables to InnoDB made the migrations work.  I pulled a recent backup, updated the "ENGINE=MyISAM" in each "CREATE TABLE" statement to read "ENGINE=InnoDB", then dropped and rebuilt database from backup, and now the migrations work as expected.  Not sure what I should do with this.  Is this something I should report as a bug somewhere?  Seems odd that MyISAM tables cause migrations to break.

**Comment by jonathanmorgan** [comment]
Last detail - after I got the tables upgraded to InnoDB, I was able to just remove the rows for photologue from django_migrations, then backup and remove the photologue_* tables, then re-run the migrations and everything ran through just fine.  Restored information to tables and back in business.  Not sure if this is a good idea for everyone, but it worked for me.

**Comment by richardbarran** [comment]
Thanks for reporting back on the results of your investigation. I know relatively little about MySQL, so I can't add anything to what you've already discovered.

I notice that myISAM tables do not support foreign key constraints, which I find a bit strange - InnoDB tables *do* support FKs, so shouldn't Django projects default to using InnoDB? I don't know.

I will close this ticket now as you've fixed the issue, but it's good to have it here in the archives in case anyone else encounters the same issue.

**Comment by jonathanmorgan** [comment]
Just for reference, the MyISAM vs. InnoDB choice is in MySQL, and it has defaulted to InnoDB since 5.5 was released (general release December, 2010).  These are pretty old databases - we started using django with this client and application in spring of 2010, django 1.2.  And, they were running on CentOS 5 originally, so I think it was running MySQL 5.0 (and it might have been CentOS 4.X before, and so an even older version of MySQL).  So, you probably wouldn't run into this problem very freq
